controllers/db/userroles.controller.ts

```diff
@@ -0,0 +1,45 @@
+import sequelize from "../../src/db/db";
+
+
+export const findUserPermissions = async (attrValue: string, attrType: string) => {
+
+    let userRolesList: any = [];
+    try {
+        if (attrType === "email") {
+            userRolesList = await sequelize.query(
+                "SELECT au.personIdentifier, roleLabel FROM admin_users as au INNER JOIN admin_users_roles as aur " +
+                "ON au.userID = aur.userID INNER JOIN admin_roles ar ON aur.roleID = ar.roleID     WHERE     au.email = :email",
+                {
+                    replacements: { email: attrValue },
+                    nest: true,
+                    raw: true
+                }
+            );
+            return JSON.stringify(userRolesList);
+        } else {
+            userRolesList = await sequelize.query(
+                "SELECT au.personIdentifier, roleLabel FROM admin_users as au INNER JOIN admin_users_roles as aur " +
+                "ON au.userID = aur.userID INNER JOIN admin_roles ar ON aur.roleID = ar.roleID     WHERE     au.personIdentifier = :personIdentifier",
+                {
+                    replacements: { personIdentifier: attrValue },
```

Comment:
Is the only thing different between these two conditions just the replacements field. If so refactor this? This doesn't need to be an if/else statement repeating 95% of the same exact code

Reply:
will consider adding scopes in future release.